src/nupic/algorithms/Connections.hpp

```diff
@@ -410,6 +419,16 @@ class Connections : public Serializable
                     const std::vector<Permanence> &previousUpdates,
                           std::vector<Permanence> &currentUpdates);
 
+  void adaptSegment(const Segment segment,
+                    const sdr::SDR &inputs,
+                    const Permanence increment,
+                    const Permanence decrement)
+  {
+    std::vector<Permanence> previousUpdates, currentUpdates;
+    adaptSegment( segment, inputs, increment, decrement,
```

Comment:
similar change as in #371 , instead of adding an overload of the method, use the one with full args, but provide defaults. 

Also, this call has performance penalty, the logic of prevUpdates is not used, but currUpdates are still stored (not needed)

Reply:
The compiler will not let me mix default arguments and references...

Reply:
> The compiler will not let me mix default arguments and references.. 

we could use pointers, but maybe the current with 2 overloads is a better approach